
<launch>
    
  
  <group if="$(eval mode==1)">
  <node name="MyRobotHardwareInterface" pkg="mini6dof_arm" type="MyRobot_hardware_interface" output="screen" ns="/arm_robot"/>  
  </group>
  <group if="$(eval moveit==false)">
  <rosparam file="$(find mini6dof_arm)/config/controllers.yaml" command="load"/>
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/arm_robot"
        args="
            joint_state_controller
            joint1_position_controller
            joint2_position_controller
            joint3_position_controller
            joint4_position_controller
            joint5_position_controller
            joint6_position_controller
        "/>
  </group>


  <group if="$(eval moveit==true)">
  <rosparam file="$(find mini6dof_arm)/config/moveit_controllers.yaml" command="load"/>
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/arm_robot"
    args="
        joint_state_controller
        arm_controller
    "/>
  </group>

  <group if="$(eval mode==2 and pid==false)">
  <node name="simple_moveit_subscriber" pkg="mini6dof_arm" type="simple_moveit_subscriber" respawn="false" required="false" output="screen" args="false"/>
  </group>

  <group if="$(eval mode==2 and pid==true)">
  <node name="simple_moveit_subscriber" pkg="mini6dof_arm" type="simple_moveit_subscriber" respawn="false" required="false" output="screen" args="true"/>
  </group>  

  <group if="$(eval pid==true)">
  <include file="$(find dynamixel_control)/launch/reconfigure_PID.launch"/>
  </group>


</launch>

